modules/admin-quota-limits.adoc

```diff
@@ -0,0 +1,392 @@
+
+|`MaxLimitRequestRatio`
+|`MaxLimitRequestRatio[resource]` less than or equal to (`container.resources.limits[resource]` / `container.resources.requests[resource]`).
+
+|===
+
+== Image Limits
+
+
+*Supported Resources:*
+
+* Storage
+
+*Resource type name:*
+
+- `openshift.io/Image`
+
+Per image, the following must hold true if specified:
+
+.Image
+[cols="3a,8a",options="header"]
+|===
+|Constraint |Behavior
+
+|`Max`
+|`image.dockerimagemetadata.size` less than or equal to `Max[resource]`
+|===
+
+ifdef::openshift-enterprise,openshift-origin[]
+[NOTE]
+====
+To prevent blobs that exceed the limit from being uploaded to the registry, the registry must be configured to enforce quota. The `REGISTRY_MIDDLEWARE_REPOSITORY_OPENSHIFT_ENFORCEQUOTA` environment variable must be set to `true`. By default, the environment variable is set to `true` for new deployments.
+====
+endif::[]
+
+[WARNING]
+====
+The image size is not always available in the manifest of an uploaded image. This is especially the case for images built with Docker 1.10 or higher and pushed to a v2 registry. If such an image is pulled with an older Docker daemon, the image manifest is converted by the registry to schema v1 and does not include all the size information. No storage limit set on images will prevent it from being uploaded.
+
+link:https://github.com/openshift/origin/issues/7706[The issue] is being addressed.
+====
+
+
+== Image Stream Limits
+
+
+*Supported Resources:*
+
+* `openshift.io/image-tags`
+* `openshift.io/images`
+
+*Resource type name:*
+
+- `openshift.io/ImageStream`
+
+Per image stream, the following must hold true if specified:
+
+.ImageStream
+[cols="3a,8a",options="header"]
+|===
+|Constraint |Behavior
+
+|`Max[openshift.io/image-tags]`
+|`length( uniqueimagetags( imagestream.spec.tags ) )` less than or equal to `Max[openshift.io/image-tags]`
+
+`uniqueimagetags` returns unique references to images of given spec tags.
+
+|`Max[openshift.io/images]`
+|`length( uniqueimages( imagestream.status.tags ) )` less than or equal to `Max[openshift.io/images]`
+
+`uniqueimages` returns unique image names found in status tags. The name is equal to the digest for the image.
+
+|===
+
+== Counting of Image References
+
+The `openshift.io/image-tags` resource represents unique xref:../nodes/clusters/nodes-cluster-limit-ranges.adoc#nodes-cluster-limit-stream-limits[stream limits]. Possible references are an `ImageStreamTag`, an `ImageStreamImage`, or a `DockerImage`. Tags can be created by using the `oc tag` and `oc import-image` commands or by using xref:../openshift_images/image-streams-manage.adoc#images-imagestream-use_image-streams-managing[managing images streams]. No distinction is made between internal and external references. However, each unique reference that is tagged in an image stream specification is counted just once. It does not restrict pushes to an internal container image registry in any way, but is useful for tag restriction.
```

Comment:
@snarayan-redhat I saw this and elected to leave it alone, because I'm afraid I'll lose important information if I move it all the way down to "Additional resources". This document is in response to a customer case and I think that specific paragraph might speak to the problem the customer ran into.

My plan is to rewrite this section such that the important facts are captured in a paragraph and the xrefs are moved from that paragraph to an "Additional resources section immediately below the paragraph.

Reply:
I understand what you're trying to do  @GroceryBoyJr but it breaks on the customer portal so we can't merge it like this. You'll have to create Additional resources sections underneath each module at the assembly level. The only way to get this to show up exactly where you want it is to break this module into smaller modules so that you can have an additional resources at the assembly between them.